conformance/lang/expressions/logical-expr/logical_and.balt

```diff
@@ -0,0 +1,699 @@
+    boolean _ = true && (a == "text"? 2 : 1); // @error static type of each expression must be a subtype of boolean
+    boolean _ = false && (a == "text"? 2 : 1); // @error static type of each expression must be a subtype of boolean
+    boolean _ = (a == "text"? 2 : 1) && false; // @error static type of each expression must be a subtype of boolean
+    boolean _ = (a == "text"? 2 : 1) && (a == "text"? 2 : 1); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using checking expression.
+Labels: logical-expr, checking-expr, int:fromString
+
+function errorFunction() {
+    boolean _ = true && check int:fromString("123"); // @error static type of each expression must be a subtype of boolean
+    boolean _ = check int:fromString("123") && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = check int:fromString("123") && check int:fromString("123"); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using trap expression.
+Labels: logical-expr, trap-expr, multiplicative-expr
+
+function errorFunction() {
+    boolean _ = true && trap divideNum(1, 0); // @error static type of each expression must be a subtype of boolean
+    boolean _ = divideNum(1, 0) && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = divideNum(1, 0) && divideNum(1, 0); // @error static type of each expression must be a subtype of boolean
+}
+
+function divideNum(int a, int b) returns int {
+    return a / b;
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using xml navigate expression.
+Labels: logical-expr, xml-step-expr, xml, xml-template-expr
+
+function errorFunction() {
+    xml a = xml `<root><ele>true</ele><ele>false</ele></root>`;
+    boolean _ = true && a/*; // @error static type of each expression must be a subtype of boolean
+    boolean _ = a/* && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = a/* && a/*; // @error static type of each expression must be a subtype of boolean
+    
+    boolean _ = true && a/<ele>; // @error static type of each expression must be a subtype of boolean
+    boolean _ = a/<ele> && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = a/<ele> && a/<ele>; // @error static type of each expression must be a subtype of boolean
+
+    boolean _ = true && a/<ele>/*; // @error static type of each expression must be a subtype of boolean
+    boolean _ = a/<ele>/* && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = a/<ele>/* && a/<ele>/*; // @error static type of each expression must be a subtype of boolean
+}
+
```

Comment:
Can you please point to the test(s) that test the static type of the logical expressions being boolean?

Reply:
We should ideally have tests with var too?